From da206ed2570fbce4d74f8a134d016d9ee99efb82 Mon Sep 17 00:00:00 2001
From: Long Dang <long.dang.aj@rvc.renesas.com>
Date: Fri, 12 Aug 2016 10:31:38 +0700
Subject: [PATCH 36/42] gst145-omxvideodec: Fix an incorrect plane size
 calculation for exporting dmabuf descriptors

Porting from a patch in Linux 3.10 environment made by Kazunori Kobayashi <kkobayas@igel.co.jp>

This change takes account into a subsampling value of a height
in order to calculate second and subsequent planes.

Signed-off-by: Long Dang <long.dang.aj@rvc.renesas.com>
---
 omx/gstomxbufferpool.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/omx/gstomxbufferpool.c b/omx/gstomxbufferpool.c
index 28effb0..033a39c 100644
--- a/omx/gstomxbufferpool.c
+++ b/omx/gstomxbufferpool.c
@@ -650,7 +650,8 @@ gst_omx_buffer_pool_acquire_buffer (GstBufferPool * bpool,
               omx_buf->pOutputPortPrivate;
   
           phys_addr = (guint) decode_res->pvPhysImageAddressY + vmeta->offset[i];
-          plane_size[i] = vmeta->stride[i] * vmeta->height;
+          plane_size[i] = vmeta->stride[i] *
+              GST_VIDEO_INFO_COMP_HEIGHT (&pool->video_info, i);
   
           res =
               mmngr_export_start_in_user (&vdbuf_data->id_export[i],
-- 
1.9.1

